ambry-router/src/main/java/com/github/ambry/router/SimpleOperationTracker.java

```diff
@@ -248,6 +248,18 @@ class SimpleOperationTracker implements OperationTracker {
         // Here set the success target to the number of eligible replicas.
         replicaSuccessTarget = eligibleReplicas.size();
         break;
+      case ReplicateBlobOperation:
+        // Replicate one blob. Use the same "success target" and "request parallelism" config settings as PutOperation.
+        // Only difference is for ReplicateBlobOperation, crossColoEnabled is true.
+        // ON_DEMAND_REPLICATION_TODO: we may consider add its own configure setting if needed.
+        eligibleReplicas = getEligibleReplicas(datacenterName, EnumSet.of(ReplicaState.STANDBY, ReplicaState.LEADER));
+        replicaSuccessTarget =
+            routerConfig.routerGetEligibleReplicasByStateEnabled ? Math.max(eligibleReplicas.size() - 1,
```

Comment:
We should treat ReplicateBlob as a special case for replica pool. If we are here, then it's likely that two replicas in colo datacenter are down. We should probably prioritize remote datacenter in the replica pool.

Reply:
Can we fine tune it later? For simplicity, for now, we pick the local first and then remote. I've added TODO in the second revision.